productmapping/src/main/java/fi/hel/verkkokauppa/productmapping/service/product/ProductMappingService.java

```diff
@@ -37,27 +37,27 @@ public ProductMapping findById(String productId) {
         return null;
     }
 
-    public ProductMapping createByParams(String namespace, String namespaceEntityId) {
+    public ProductMapping createByParams(String namespace, String namespaceEntityId, String merchantId) {
         String productId = UUIDGenerator.generateType3UUIDString(namespace, namespaceEntityId);
```

Comment:
I was thinking should you add merchantId to type3UUID too, because this prevents different merchants to have unique productId:s.

String productId = UUIDGenerator.generateType3UUIDString(namespace, namespaceEntityId);
`String productId = UUIDGenerator.generateType3UUIDString(productId , namespace);`

This change should be tested when creating new or changing values of mapping.

Reply:
Okay so did you mean that it should be:
String productId = UUIDGenerator.generateType3UUIDString(merchantId , namespace);